server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1090,6 +1090,9 @@ private ReplicationGroup calculateReplicationGroup() {
         }
 
         assert indexSettings().isRemoteTranslogStoreEnabled()
+            // Handle migration cases. Ignore assertion if any of the shard copies in the replication group is assigned to a remote node
+            || (replicationGroup != null
+                && replicationGroup.getReplicationTargets().stream().anyMatch(ShardRouting::isAssignedToRemoteStoreNode))
```

Comment:
can be simplified by just using `indexSettings.isRemoteNode()` instead of 

```
indexSettings().isRemoteTranslogStoreEnabled()
            || indexSettings.isRemoteNode()
            // Handle migration cases. Ignore assertion if any of the shard copies in the replication group is assigned to a remote node
            || (replicationGroup != null
                && replicationGroup.getReplicationTargets().stream().anyMatch(ShardRouting::isAssignedToRemoteStoreNode))
```